R/expect_s3_class_linter.R

```diff
@@ -0,0 +1,108 @@
+#' Require usage of expect_s3_class()
+#'
+#' [testthat::expect_s3_class()] exists specifically for testing the class
+#'   of S3 objects. [testthat::expect_equal()], [testthat::expect_identical()],
+#'   and [testthat::expect_true()] can also be used for such tests,
+#'   but it is better to use the tailored function instead.
+#'
+#' @evalRd rd_tags("expect_s3_class_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+expect_s3_class_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$parsed_content) == 0L) {
+      return(list())
+    }
+
+    xml <- source_file$xml_parsed_content
+
+    # (1) expect_{equal,identical}(class(x), C)
+    # (2) expect_true(is.<class>(x)) and expect_true(inherits(x, C))
+    is_class_call <- xp_text_in_table(c(is_s3_class_calls, "inherits"))
```

Comment:
false positive lint caused by glue: local variable ‘is_class_call’ assigned but may not be used.